### DISCUSSION TITLE
About Soil various problem.

### DISCUSSION THREAD

**Discussion by FuYingWang** [NCAR/noahmp#160] [General]
Hello,
After I completed a simple simulation, I worked on improving the simulation accuracy. I improved the 'HGT_M' and 'LANDUSEF' variables in the 'geo_em.d01.nc' file using more accurate data. Next, I'm going to use more precise soil variables instead of the defaults, but I'm running into problems.
1. there seem to be only a few variables I've found regarding soil  (e.g.CLAYFRAC, SOILCBOT,SOILCL1). 
**_I'd like to know if noahmp is using the 'CLAYFRAC' and 'SOILCBOT' variables??_.**
![soiltype](https://github.com/user-attachments/assets/1ba78f38-eef3-4d1f-8298-472e737b6a6e)
2.I have more precise data for soil variables  (such as sand%, clay%, soil classification), how should I use them to replace the default values.
Thanks.

**Comment by tslin2** [comment]
There are options in the namelist.hrldas to change:
 SOIL_DATA_OPTION                  = 1,                      ! Noah-MP options for defining soil properties   [default = 1]
                                                             ! geogrid must have been run with GEOGRID.TBL.ARW.noahmp, use with caution
                                                             ! **1 -> use input dominant soil texture
                                                             !   2 -> use input soil texture that varies with depth
                                                             !   3 -> use soil composition (sand, clay, orgm) and pedotransfer functions (OPT_PEDO)
                                                             !   4 -> use input soil properties (BEXP_3D, SMCMAX_3D, etc.) (not valid in WRF)

Default option 1 use top layer soil texture for all soil layers in the Noah-MP
Option 2 use SOILC1-4, so soil texture change with soil layer.
Option 3 use clay and sand % from SOILCOMP, it has 8 values.
1-4 is sand % for each 4 soil layer, 5-8 is clay % for each 4 soil layer 

You could use option 3 to change the SOILCOMP  from default values to your soil data

**Reply by FuYingWang** [reply]
Thanks a lot.
I have add soil variable in geo_em.d01.nc about ' SOILCOMP'. And ncview can read it.
![image](https://github.com/user-attachments/assets/743baa74-7dda-4dfb-8302-e71ecf5bc235)
After creating GLDAS_forcing and change SOIL_DATA_OPTION = 3, I run the model.
But, When I run _./hrldas.exe namelist.hrldas_, an error apper.
![image](https://github.com/user-attachments/assets/be71a3a5-990d-4550-aa62-0f0beb8f9675)
Then, I _ncview HRLDAS_setup_2023010100_d1_
![image](https://github.com/user-attachments/assets/51190bbe-a2ad-48b0-9755-9f37f01894dc)
It don't have SOILCOMP.
**_What details did I miss? And what should I do_**

**Comment by FuYingWang** [comment]
Thanks a lot.
I have add soil variable in geo_em.d01.nc about ' SOILCOMP'. And ncview can read it.
image
After creating GLDAS_forcing and change SOIL_DATA_OPTION = 3, I run the model.
But, When I run ./hrldas.exe namelist.hrldas, an error apper.
![image](https://github.com/user-attachments/assets/d62275a1-8ac4-4409-9a96-d5dab9227599)
Then, I ncview HRLDAS_setup_2023010100_d1
![image](https://github.com/user-attachments/assets/dee85d24-fc25-4252-abab-a8cd82516c7c)
It don't have SOILCOMP.

What details did I miss? And what should I do

**Reply by cenlinhe** [reply]
please directly add SOILCOMP to HRLDAS_setup file. Our current HRLDAS_forcing preprocessor does not have this SOILCOMP variable, which will not pass it from geo_em file to HRLDAS_setup file. We will add it to HRLDAS_forcing package later. Thanks!

**Reply by FuYingWang** [reply]
Sorry to bother you again, I followed your instructions and added 'SOILCOMP' to the 'HRLDA_SETUP' file. 
SOILCOMP, it has 8 values. 1-4 is sand % for each 4 soil layer, 5-8 is clay % for each 4 soil layer. 
But, the 'PedoTransferSR2006Mod.F90' seem also need _**'Orgm'**_, I only have sand and clay.
I can run ./rrldas.exe, but I don't seem to have 'Orgm' data. what would happen? What will be the value of Orgm?
Thanks.

**Reply by tslin2** [reply]
This assumes zero in the current model, it is not necessary to have it. 
https://github.com/NCAR/noahmp/blob/fa2d2098bac1e437a21ccc15494b152996d5bc84/drivers/hrldas/WaterVarInTransferMod.F90#L230

if you like to replace it, you will revise this value. Or you can revise the code to read other values
